src/browser/telemetry.js

```diff
@@ -0,0 +1,228 @@
+var _ = require('../utility');
+var urlparser = require('./url');
+
+var defaults = {
+  network: true,
+  log: true,
+  dom: true,
+  navigation: true
+};
+
+function replace(obj, name, replacement, replacements) {
+  var orig = obj[name];
+  obj[name] = replacement(orig);
+  if (replacements) {
+    replacements.push([obj, name, orig]);
+  }
+}
+
+function restore(replacements) {
+  var b;
+  while (replacements.length) {
+    b = replacements.shift();
+    b[0][b[1]] = b[2];
+  }
+}
+
+function Instrumenter(options, telemeter, rollbar, _window, _document) {
+  var autoInstrument = options.autoInstrument;
+  if (autoInstrument === false) {
+    this.autoInstrument = {};
+    return;
+  }
+  if (!_.isType(autoInstrument, 'object')) {
+    autoInstrument = defaults;
```

Comment:
Can we log a warning or something here? It could help debugging configuration issues.